src/components/draw/DrawDirective.js

```diff
@@ -327,110 +559,227 @@ goog.require('ga_map_service');
-          scope.$watch('options.isModifyActive', function(active) {
-            if (active) {
-              activateModifyInteraction();
+
+          ////////////////////////////////////
+          // Tab managment
+          ////////////////////////////////////
+          scope.activeTabProps = function() {
+            scope.isPropsActive = true;
+            scope.isMeasureActive = false;
+            scope.options.isProfileActive = false;
+          };
+          scope.activeTabProfile = function() {
+            scope.isPropsActive = false;
+            scope.isMeasureActive = false;
+            scope.options.isProfileActive = true;
+          };
+          scope.activeTabMeasure = function() {
+            scope.isPropsActive = false;
+            scope.isMeasureActive = true;
+            scope.options.isProfileActive = false;
+          };
+          scope.showTabs = function(feature) {
+            var bools = [
+              scope.showPropsTab(feature),
+              scope.showMeasureTab(feature),
+              scope.showProfileTab(feature)
+            ];
+            var cpt = 0;
+            for (var i in bools) {
+              if (bools[i]) {
+                 cpt++;
+              }
             }
-          });
-          scope.$watch('options.isDeleteActive', function(active) {
-            if (active) {
-              deleteAllFeatures();
-              scope.options.isDeleteActive = false;
+            return (cpt > 1);
+          };
+          scope.showMeasureTab = function(feature) {
+            var geom = feature.getGeometry();
+            var isPoint = (geom instanceof ol.geom.Point);
+            return (feature.get('type') != 'measure' && !isPoint);
+          };
+          scope.showProfileTab = function(feature) {
+            return scope.showMeasureTab(feature);
+          };
+          scope.showPropsTab = function(feature) {
+            var geom = feature.getGeometry();
+            var isPoint = (geom instanceof ol.geom.Point);
+            if (isPoint) {
+              scope.activeTabProps();
             }
-          });
+            return (!isPoint && feature.get('type') != 'measure');
+          };
+
+
+
 
+          ////////////////////////////////////
           // create/update the file on s3
-          var save = function() {
+          ////////////////////////////////////
+          var save = function(evt) {
+            if (layer.getSource().getFeatures().length == 0) {
+              //if no features to save do nothing
+              return;
+            }
+            scope.statusMsgId = 'draw_file_saving';
             var kmlString = gaExportKml.create(layer,
                 map.getView().getProjection());
             var id = layer.adminId ||
                 gaFileStorage.getFileIdFromFileUrl(layer.url);
             gaFileStorage.save(id, kmlString,
                 'application/vnd.google-earth.kml+xml').then(function(data) {
+              scope.statusMsgId = 'draw_file_saved';
+
```

Comment:
Mentioned it already when introducing kml saving. I think we should alert the user if somethign goes wrong when trying to save the kml (might be rare, I know). Maybe send directly an email to us :)